Include/pystate.h

```diff
@@ -48,6 +48,9 @@ typedef struct _is {
     PyObject *import_func;
     /* Initialized to PyEval_EvalFrameDefault(). */
     _PyFrameEvalFunction eval_frame;
+
+    Py_ssize_t co_extra_user_count;
+    freefunc co_extra_freefuncs[MAX_CO_EXTRA_USERS];
 } PyInterpreterState;
```

Comment:
Similar to the issue with `PyThreadState` below, this changes the size of `PyInterpreterState`, and that's a public struct, and hence may have fixed size storage allocated in extension modules. That's significantly less likely to be the case for `PyInterpreterState` than it is for `PyThreadState`, but it's still an ABI break.

About the only way I see to avoid that is to to consider these two fields to be part of the interpreter state, and manipulate them accordingly, but leave the *actual storage* (at least for 3.6) inside `tstate_head` (copying them between threads states as necessary to cope with that indirection).

And then for 3.7, you can do this the simple way.